### PR TITLE
Issue 18216 integrate dot key value into dot apps

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/view/components/dot-key-value/dot-key-value-table-input-row/dot-key-value-table-input-row.component.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+     */
+    saveVariable(): void {
+        this.save.emit(this.variable);
+        this.cleanVariableValues();
+        this.keyCell.nativeElement.focus();
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    private getElementToFocus($event: KeyboardEvent): void {
+        if (DotKeyValueUtil.isKeyInput($event) || this.variable.value === '') {
```
Comment (fmontes) [inline src/app/view/components/dot-key-value/dot-key-value-table-input-row/dot-key-value-table-input-row.component.ts:131]: Can you move this out to its own func?

File: src/app/view/components/dot-key-value/dot-key-value.component.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        this.save.emit(variable);
+
+        variable = this.setHiddenValue(variable);
+        const indexChanged = DotKeyValueUtil.getVariableIndexChanged(variable, this.variables);
+        if (indexChanged !== null) {
+            this.variables[indexChanged] = _.cloneDeep(variable);
+        } else {
+            this.variables = [].concat(variable, this.variables);
+        }
+        this.variablesBackup = [].concat(this.variables);
```
Comment (fmontes) [inline src/app/view/components/dot-key-value/dot-key-value.component.ts:83]: spread

File: src/app/view/components/dot-key-value/dot-key-value.component.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+     * the variable to be handled by a parent smart component
+     * @param {DotKeyValue} variable
+     * @memberof DotKeyValueComponent
+     */
+    saveVariable(variable: DotKeyValue): void {
+        this.save.emit(variable);
+
+        variable = this.setHiddenValue(variable);
+        const indexChanged = DotKeyValueUtil.getVariableIndexChanged(variable, this.variables);
+        if (indexChanged !== null) {
```
Comment (fmontes) [inline src/app/view/components/dot-key-value/dot-key-value.component.ts:78]:
```suggestion
        if (indexChanged) {
```

File: src/app/portlets/dot-apps/dot-apps-configuration-detail/dot-apps-configuration-detail.component.ts [hunk 4]
```diff
@@ -76,14 +73,62 @@ export class DotAppsConfigurationDetailComponent implements OnInit {
...
     }
 
+    /**
+     * Handle Save event doing if new a prepend, otherwise a replace
+     * to the local collection
+     * @param {DotKeyValue} variable
+     * @memberof DotAppsConfigurationDetailComponent
+     */
+    saveDynamicVariable(variable: DotKeyValue): void {
+        const indexChanged = DotKeyValueUtil.getVariableIndexChanged(variable, this.dynamicVariables);
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply src/app/portlets/dot-apps/dot-apps-configuration-detail/dot-apps-configuration-detail.component.ts:83]: I prefer not, the DotKeyValue should only IMO care about sending the variable that was saved/updated

**Pull request merged**